### PR TITLE
S3 buckets for participant files API

### PR DESCRIPTION
**Pull request by alxdarksage**
...

### CODE REVIEW

**Comment by DwayneJengSage** [comment]
The example we used for creating a bucket for the Exporter 3.0 stuff https://github.com/Sage-Bionetworks/BridgeServer2-infra/pull/32

We don't need a bucket policy for Participant Files, since we don't need to expose it outside of the Bridge AWS account.

Also, we have an existing open PR for Participant S3 bucket https://github.com/Sage-Bionetworks/BridgeServer2-infra/pull/36

File: config/uat/bridgeserver2.yaml
Comment (alxdarksage) [inline config/uat/bridgeserver2.yaml]: Should I remove these? Aren't the bucket names now basically randomly generated by CloudFormation?

File: templates/bridgeserver2.yaml
Comment (alxdarksage) [inline templates/bridgeserver2.yaml]: Khai says this is necessary.

Comment (zaro0508) [reply templates/bridgeserver2.yaml]: I see that we are giving beanstalk instances full s3 bucket access therefore the bridge app should have access to this bucket via the [AWSIAMRole](https://github.com/Sage-Bionetworks/BridgeServer2-infra/blob/develop/templates/bridgeserver2.yaml#L540).  In that case this policy is not needed.

from your commit message `... I think all the permissions we wish to extend through a resigned URL have to be available to the AWS client`.  I'm not sure what you mean by that but if your goal is to make the contents of this bucket available from AWS CLI  then you may need an extra set of policies to make that happen.

File: templates/bridgeserver2.yaml
Comment (zaro0508) [inline templates/bridgeserver2.yaml:390]: CFN will create a bucket with a random name then this !Ref will return that bucket name.  Therefore you don't need to pass in the bucket name.

File: config/uat/bridgeserver2.yaml
Comment (zaro0508) [reply config/uat/bridgeserver2.yaml]: yes, you don't need to pass in the name you are already referring to it.

**Review by zaro0508 — CHANGES_REQUESTED** [review]

File: templates/bridgeserver2.yaml
Comment (alxdarksage) [reply templates/bridgeserver2.yaml]: Our code will return temporary presigned URLs through the API to upload a file to this bucket, and to later download the file from this bucket. I had read that the AWS client that asks for these URLs cannot assign them more privileges than it itself possesses, but it appears our AWS S3 client has full S3 bucket access.

Comment (DwayneJengSage) [reply templates/bridgeserver2.yaml]: We don't need this. This particular bucket policy grants access to Synapse. Since we don't want to expose this to Synapse, we don't want to include this bucket policy.

File: config/develop/bridgeserver2.yaml
Comment (zaro0508) [inline config/develop/bridgeserver2.yaml]: you need to remove this

File: config/uat/bridgeserver2.yaml
Comment (zaro0508) [inline config/uat/bridgeserver2.yaml]: remove this

File: templates/bridgeserver2.yaml
Comment (zaro0508) [inline templates/bridgeserver2.yaml]: remove this

**Review by zaro0508 — CHANGES_REQUESTED** [review]

File: config/develop/bridgeserver2.yaml
Comment (alxdarksage) [reply config/develop/bridgeserver2.yaml]: removed

File: config/uat/bridgeserver2.yaml
Comment (alxdarksage) [reply config/uat/bridgeserver2.yaml]: removed

File: templates/bridgeserver2.yaml
Comment (alxdarksage) [reply templates/bridgeserver2.yaml]: removed (okay I get what it's doing now... it's setting an environment variable to the name of the bucket resource)

File: templates/bridgeserver2.yaml
Comment (alxdarksage) [reply templates/bridgeserver2.yaml]: (removed)

File: templates/bridgeserver2.yaml
Comment (zaro0508) [inline templates/bridgeserver2.yaml]: please fix your indentation.

**Review by zaro0508 — CHANGES_REQUESTED** [review]

File: templates/bridgeserver2.yaml
Comment (alxdarksage) [reply templates/bridgeserver2.yaml]: fixed

**Review by DwayneJengSage — APPROVED** [review]
Looks good to me. However, let's hold off on merging this until after we resolve the CloudFormation conversation.

**Comment by alxdarksage** [comment]
Dan has been waiting for this particular PR for over a month. I'm going to merge this when it's verified by Khai so he can move forward finally.

**Comment by DwayneJengSage** [comment]
I think we can afford to wait one day. If after Thursday's meeting, we still don't have a resolution on the CloudFormation issue, then we can go ahead and merge to unblock Dan.

What I want to avoid is that we merge this in, and then tomorrow, we immediately back out of the change because we decide we don't want to add things to CloudFormation anymore.

**Comment by DwayneJengSage** [comment]
As mentioned in today's meeting, we're going to back out of this change, manually create the participant file buckets to unblock the feature, and long term our S3 buckets will be tracked by our solution to https://sagebionetworks.jira.com/browse/BRIDGE-2909

**Pull request closed without merging**
